Sources/FoundationEssentials/JSON/JSONDecoder.swift

```diff
@@ -341,6 +341,24 @@ open class JSONDecoder {
     /// - throws: `DecodingError.dataCorrupted` if values requested from the payload are corrupted, or if the given data is not valid JSON.
     /// - throws: An error if any value throws an error during decoding.
     open func decode<T: Decodable>(_ type: T.Type, from data: Data) throws -> T {
+        try _decode({
+            try $0.unwrap($1, as: type, for: .root, _JSONKey?.none)
+        }, from: data)
+    }
+    
+    @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+    open func decode<T: DecodableWithConfiguration>(_ type: T.Type, from data: Data, configuration: T.DecodingConfiguration) throws -> T {
+        try _decode({
+            try $0.unwrap($1, as: type, configuration: configuration, for: .root, _JSONKey?.none)
+        }, from: data)
+    }
+    
+    @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+    open func decode<T: DecodableWithConfiguration, C : DecodingConfigurationProviding>(_ type: T.Type, from data: Data, configuration: C.Type) throws -> T where T.DecodingConfiguration == C.DecodingConfiguration {
```

Comment:
Nit: the generic argument list has names so long that I wonder whether it would be better to have just `<T, C> ` with everything else in the `where` clause.